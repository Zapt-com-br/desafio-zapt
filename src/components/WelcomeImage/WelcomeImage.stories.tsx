import React from 'react';

import { Story, Meta } from '@storybook/react/types-6-0';
import WelcomeImage from '.';

export default {
  title: 'Welcome/WelcomeImage',
  component: WelcomeImage,
} as Meta;

export const WelcomeImageDefault: Story = () => <WelcomeImage />;
