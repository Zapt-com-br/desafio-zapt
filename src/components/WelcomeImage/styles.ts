import styled from 'styled-components';

import image from 'assets/image.svg';

export const Wrapper = styled.div`
  background: #ffff url(${image}) no-repeat;
  width: 100%;
  min-height: 31rem;
  max-width: 41rem;
  position: relative;
`;
export const Image = styled.img`
  width: 100%;
`;
export const BowWrapper = styled.div`
  position: absolute;
  bottom: 4rem;
  right: 4rem;
`;
export const CircleWrapper = styled.div`
  position: absolute;
  top: 1.6rem;
  left: 7rem;
`;
