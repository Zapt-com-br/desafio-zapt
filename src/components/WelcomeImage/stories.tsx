import { Story, Meta } from '@storybook/react/types-6-0';
import WelcomeImage from '.';

export default {
  title: 'WelcomeImage',
  component: WelcomeImage,
} as Meta;

export const Default: Story = () => <WelcomeImage />;
