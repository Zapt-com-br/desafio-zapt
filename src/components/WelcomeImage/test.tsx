import { render, screen } from 'utils/test-utils';

import WelcomeImage from '.';

describe('<WelcomeImage />', () => {
  it('should render correctly', () => {
    const { container } = render(<WelcomeImage />);

    expect(
      screen.getByRole('img', { name: /grupo de amigos/i })
    ).toBeInTheDocument();

    expect(screen.getByRole('img', { name: /circle/i })).toBeInTheDocument();
    expect(
      screen.getByRole('img', { name: /quarter circumference/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
