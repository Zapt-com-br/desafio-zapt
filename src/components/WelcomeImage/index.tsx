import React from 'react';
import * as Styled from './styles';

import bow from 'assets/bow_image.svg';
import circleImg from 'assets/circle_img.svg';

const WelcomeImage = () => (
  <Styled.Wrapper>
    <Styled.CircleWrapper>
      <Styled.Image src={circleImg} alt="circle" />
    </Styled.CircleWrapper>
    <Styled.BowWrapper>
      <Styled.Image src={bow} alt="bow" />
    </Styled.BowWrapper>
  </Styled.Wrapper>
);

export default WelcomeImage;
