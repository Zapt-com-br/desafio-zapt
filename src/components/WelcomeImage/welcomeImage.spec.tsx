import React from 'react';
import { screen } from '@testing-library/react';
import { renderWithTheme } from 'helpers/renderWithThame';

import WelcomeImage from '.';

import image from 'assets/image.svg';

describe('<WelcomeImage />', () => {
  it('should render WelcomeImage', () => {
    const { container } = renderWithTheme(<WelcomeImage />);

    const bow = screen.getByRole('img', { name: 'bow' });
    const circle = screen.getByRole('img', { name: 'circle' });

    expect(bow).toBeInTheDocument();
    expect(circle).toBeInTheDocument();
    expect(container.firstChild).toHaveStyleRule(
      'background',
      `#ffff url(${image}) no-repeat`
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
