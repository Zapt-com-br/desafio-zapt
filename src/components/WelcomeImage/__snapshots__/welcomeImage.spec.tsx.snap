// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomeImage /> should render WelcomeImage 1`] = `
.c0 {
  background: #ffff url(image.svg) no-repeat;
  width: 100%;
  min-height: 31rem;
  max-width: 41rem;
  position: relative;
}

.c2 {
  width: 100%;
}

.c3 {
  position: absolute;
  bottom: 4rem;
  right: 4rem;
}

.c1 {
  position: absolute;
  top: 1.6rem;
  left: 7rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="circle"
      class="c2"
      src="circle_img.svg"
    />
  </div>
  <div
    class="c3"
  >
    <img
      alt="bow"
      class="c2"
      src="bow_image.svg"
    />
  </div>
</div>
`;
